src/cloud/components/onboarding/EngagementLink.tsx

```diff
@@ -18,16 +18,20 @@ const EngagementLink: FC = () => {
   const org = useSelector(getOrg)
   const me = useSelector((state: AppState) => state.me)
 
-  const {usageDataStates} = useGetUserStatus()
   useEffect(() => {
-    if (userpilot) {
-      if (usageDataStates.length) {
+    if (isFlagEnabled('newUsageAPI')) {
+      handleGetUserStatus().then(({usageDataStates}) =>
         setDataStates(usageDataStates)
-      }
+      )
+    }
+  }, [])
+
+  useEffect(() => {
+    if (userpilot) {
       sendToUserPilot()
       userpilot.reload()
     }
-  }, [pathname, org, me, usageDataStates])
+  }, [pathname, org, me])
```

Comment:
for some reason my brain is telling me useCallback goes with useEffect....

Reply:
dont think its needed in this case - but i did add a memo() around the component bc it is rendered as a sibling of large parts of the main app and does not need to be rerendered every single time.